# **1주차 복습과제**
- 1주차 복습과제는 **파이썬 연습과제**입니다.
- 연습문제는 필수이고, 나머지는 필요하시면 선택적으로 실습해 보시면 됩니다.
  - 복습과제 제출은 `연습문제` 파일만 제출하시면 됩니다.
  - 제출 시 파일명 본인 이름으로 변경해 주세요.
  - 해당 파일에 문제 푸신 후 제출해 주시면 됩니다.
- 과제는 팀원들과 같이 풀이하셔도 좋고, 책을 참고해도 좋고, ChatGPT를 써도 좋고, 구글링을 해도 되고 다 좋습니다.
  - 백지만 내지 말아주세요..😥

## **1. 파이썬 자료구조**

다음과 같이 출력 되도록 코드를 작성하세요

### Q1: Type 확인

In [2]:
a = 3.14

타입을 확인해 주세요

In [3]:
print(type(a))

<class 'float'>


<p><strong>[출력 결과]</strong></p><pre>float</pre>

### Q2: Type 변경

In [4]:
a = 1

타입을 변경해 주세요

In [8]:
convert_a = float(a)
print(convert_a)

1.0


<p><strong>[출력 결과]</strong></p><pre>1.0</pre>

### Q3: Type 변경

In [9]:
a = 1

타입을 변경해 주세요

In [10]:
converted_a = bool(a)
print(converted_a)

True


<p><strong>[출력 결과]</strong></p><pre>True</pre>

### Q4: 타입 변경

In [12]:
a = False

타입을 변경해 주세요

In [13]:
converted_a = int(a)
print(converted_a)

0


<p><strong>[출력 결과]</strong></p><pre>0</pre>

### Q5: 타입 변경

In [14]:
a = 123

타입을 변경해 주세요

In [16]:
# 코드를 입력해 주세요
converted_a = str(a)
print(converted_a)

123


<p><strong>[출력 결과]</strong></p><pre>'123'</pre>

## **2. ⭐ 리스트, 튜플**

### 연습문제 1

In [23]:
movie = ['어벤져스', '아이언맨', '토르', '스파이더맨']

`movie`에 맨 끝에 `엑스맨` 을 추가하세요.

In [24]:
# 코드를 입력해 주세요
movie.append('엑스맨')
print(movie)

['어벤져스', '아이언맨', '토르', '스파이더맨', '엑스맨']


<p><strong>[출력 결과]</strong></p><pre>['어벤져스', '아이언맨', '토르', '스파이더맨', '엑스맨']</pre>

`movie`의 1번 index에 `데드풀`을 추가하세요

In [25]:
# 코드를 입력해 주세요
movie.insert(1, '데드풀')
print(movie)

['어벤져스', '데드풀', '아이언맨', '토르', '스파이더맨', '엑스맨']


<p><strong>[출력 결과]</strong></p><pre>['어벤져스', '데드풀', '아이언맨', '토르', '스파이더맨', '엑스맨']</pre>

`movie`에서 `아이언맨`을 삭제 하세요.

- remove() 사용

In [26]:
# 코드를 입력해 주세요
movie.remove('아이언맨')
print(movie)

['어벤져스', '데드풀', '토르', '스파이더맨', '엑스맨']


<p><strong>[출력 결과]</strong></p><pre>['어벤져스', '데드풀', '토르', '스파이더맨', '엑스맨']</pre>

`movie`에서 1번째 index를 삭제하세요

In [28]:
# 코드를 입력해 주세요
del movie[1]
print(movie)

['어벤져스', '토르', '스파이더맨', '엑스맨']


<p><strong>[출력 결과]</strong></p><pre>['어벤져스', '토르', '스파이더맨', '엑스맨']</pre>

In [32]:
kr_movie = ['승리호', '신세계', '타짜']

`kr_movie`를 `movie`에 추가해 주세요

In [33]:
# 코드를 입력해 주세요
movie = movie + kr_movie
print(movie)

['어벤져스', '토르', '스파이더맨', '엑스맨', '승리호', '신세계', '타짜']


<p><strong>[출력 결과]</strong></p><pre>['어벤져스', '토르', '스파이더맨', '엑스맨', '승리호', '신세계', '타짜']</pre>

`movie`를 가나다 순으로 정렬하세요

In [34]:
# 코드를 입력해 주세요
movie.sort()
print(movie)

['스파이더맨', '승리호', '신세계', '어벤져스', '엑스맨', '타짜', '토르']


<p><strong>[출력 결과]</strong></p><pre>['스파이더맨', '승리호', '신세계', '어벤져스', '엑스맨', '타짜', '토르']</pre>

### 연습문제 2

In [36]:
fruit = ['사과', '바나나', '파인애플', '배', '수박', '키위', '오렌지', '망고', '딸기']

list **인덱싱** & **슬라이싱**을 사용하여 다음과 같이 출력 하세요

In [37]:
# 코드를 입력해 주세요
result = fruit[2:5]

print(result)

['파인애플', '배', '수박']


<p><strong>[출력 결과]</strong></p><pre>['파인애플', '배', '수박']</pre>

In [39]:
# 코드를 입력해 주세요
result = fruit[7:]

print(result)

['망고', '딸기']


<p><strong>[출력 결과]</strong></p><pre>['망고', '딸기']</pre>

In [41]:
# 코드를 입력해 주세요
result = fruit[:3]

print(result)

['사과', '바나나', '파인애플']


<p><strong>[출력 결과]</strong></p><pre>['사과', '바나나', '파인애플']</pre>

In [42]:
# 코드를 입력해 주세요
result = fruit[3:]

print(result)

['배', '수박', '키위', '오렌지', '망고', '딸기']


<p><strong>[출력 결과]</strong></p><pre>['배', '수박', '키위', '오렌지', '망고', '딸기']</pre>

### 연습문제 3

슬라이싱을 활용하여 **짝수**만 출력해 주세요

In [43]:
nums = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

In [44]:
# 코드를 입력해 주세요
even_nums = [x for x in nums if x % 2 == 0]
print(even_nums)

[2, 4, 6, 8, 10]


<p><strong>[출력 결과]</strong></p><pre>[2, 4, 6, 8, 10]
</pre>

### 연습문제 4

sample = [1, 2, 3, 4, 5, 6, 7, 8] 리스트를 슬라이싱을 이용하여 출력과 같이 나올수 있도록 코드를 채워주세요.

In [45]:
sample = [1, 2, 3, 4, 5, 6, 7, 8]

슬라이싱을 사용해 다음과 같이 출력해 주세요

In [46]:
# 코드를 입력해 주세요
reversed_sample = sample[::-1]
print(reversed_sample)

[8, 7, 6, 5, 4, 3, 2, 1]


<p><strong>[출력 결과]</strong></p><pre>[8, 7, 6, 5, 4, 3, 2, 1]</pre>

슬라이싱을 사용해 다음과 같이 출력해 주세요

In [47]:
# 코드를 입력해 주세요
selected_numbers = [x for x in sample if x in [3, 5, 7]]
print(selected_numbers)

[3, 5, 7]


<p><strong>[출력 결과]</strong></p><pre>[3, 5, 7]</pre>

슬라이싱을 사용해 다음과 같이 출력해 주세요

In [48]:
# 코드를 입력해 주세요
selected_numbers = [x for x in sample if x in [2, 5, 8]]
print(selected_numbers)

[2, 5, 8]


<p><strong>[출력 결과]</strong></p><pre>[2, 5, 8]</pre>

### 연습문제 5

- 아래의 값을 저장하고 있는 `myList` 변수를 생성하세요.
- [[1,2,3],[4,5,6],[7,8]]

In [49]:
myList = [[1,2,3],[4,5,6],[7,8]]
myList

[[1, 2, 3], [4, 5, 6], [7, 8]]

- 9를 `myList`에 추가하여 아래의 값을 저장할 수 있도록 하세요.
    - [[1,2,3],[4,5,6],[7,8,9]]

In [50]:
# 코드를 입력해 주세요
myList[-1].append(9)
print(myList)

[[1, 2, 3], [4, 5, 6], [7, 8, 9]]


<p><strong>[출력 결과]</strong></p><pre>[[1, 2, 3], [4, 5, 6], [7, 8, 9]]</pre>

### 연습문제 6

주어진 tuple `a`를 (1, 100, 2, 3, 4)의 요소를 가지도록 변경해 주세요

In [51]:
a = (1, 2, 3, 4)

In [53]:
# 코드를 입력해 주세요
new_a = (1, 100, *a[1:])
print(new_a)

(1, 100, 2, 3, 4)


<p><strong>[출력 결과]</strong></p><pre>(1, 100, 2, 3, 4)</pre>

## **3. 셋, 딕셔너리**

### 연습문제 1

In [54]:
sample1 = {1, 2, 3, 4, 5}
sample2 = {2, 4, 5, 6, 7}

print(f'sample1: {sample1}')
print(f'sample2: {sample2}')

sample1: {1, 2, 3, 4, 5}
sample2: {2, 4, 5, 6, 7}


`sample1` 에 6를 추가합니다.

In [56]:
# 코드를 입력해 주세요
sample1.add(6)
print(sample1)

{1, 2, 3, 4, 5, 6}


<p><strong>[출력 결과]</strong></p><pre>{1, 2, 3, 4, 5, 6}</pre>

`sample2`에 2를 제거합니다.

In [58]:
# 코드를 입력해 주세요
sample2.remove(2)
print(sample2)

{4, 5, 6, 7}


<p><strong>[출력 결과]</strong></p><pre>{4, 5, 6, 7}</pre>

`sample1`과 `sample2`의 **교집합**을 출력합니다.

In [59]:
# 코드를 입력해 주세요
intersection_result = sample1 & sample2
print(intersection_result)

{4, 5, 6}


<p><strong>[출력 결과]</strong></p><pre>{4, 5, 6}</pre>

`sample1`과 `sample2`의 **합집합**을 출력합니다.

In [60]:
# 코드를 입력해 주세요
union_result = sample1 | sample2
print(union_result)

{1, 2, 3, 4, 5, 6, 7}


<p><strong>[출력 결과]</strong></p><pre>{1, 2, 3, 4, 5, 6, 7}</pre>

`sample1`과 `sample2`의 **차집합**을 활용하여 다음과 같이 출력합니다.

In [61]:
# 코드를 입력해 주세요
difference_result = sample1 - sample2
print(difference_result)

{1, 2, 3}


<p><strong>[출력 결과]</strong></p><pre>{1, 2, 3}</pre>

### 연습문제 2

- 다음 리스트에서 중복된 항목을 제거하세요
- 최종 출력 값은 list 형태로 출력하세요

In [62]:
numList = [1, 3, 2, 3, 7, 6, 8, 4, 10, 5, 3, 8, 9]

In [63]:
# 코드를 입력해 주세요
unique_list = list(set(numList))
print(unique_list)

[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]


<p><strong>[출력 결과]</strong></p><pre>[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]</pre>

### 연습문제 3

- 학생들의 이름과 점수를 key와 value로 가지는 `score` 변수를 생성하세요.
    - 하준 90점, 서윤 86점, 지아 80점

In [66]:
# 코드를 입력해 주세요
score = {
    "하준": 90,
    "서윤": 86,
    "지아": 80
}
print(score)

{'하준': 90, '서윤': 86, '지아': 80}


<p><strong>[출력 결과]</strong></p><pre>{'하준': 90, '서윤': 86, '지아': 80}</pre>

score 변수에 수지 95점을 추가하세요.

In [68]:
# 코드를 입력해 주세요
score["수지"] = 95
print(score)

{'하준': 90, '서윤': 86, '지아': 80, '수지': 95}


<p><strong>[출력 결과]</strong></p><pre>{'하준': 90, '서윤': 86, '지아': 80, '수지': 95}</pre>

score 변수에서 지아를 삭제하세요.

In [69]:
# 코드를 입력해 주세요
del score["지아"]
print(score)

{'하준': 90, '서윤': 86, '수지': 95}


<p><strong>[출력 결과]</strong></p><pre>{'하준': 90, '서윤': 86, '수지': 95}</pre>

score 변수에 기창 98점, 남철 60점, 기성 75점을 **한번에** 추가하세요.
- update() 를 사용하세요

In [71]:
# 코드를 입력해 주세요
score.update({
    "기창": 98,
    "남철": 60,
    "기성": 75
})
print(score)

{'하준': 90, '서윤': 86, '수지': 95, '기창': 98, '남철': 60, '기성': 75}


<p><strong>[출력 결과]</strong></p><pre>{'하준': 90, '서윤': 86, '수지': 95, '기창': 98, '남철': 60, '기성': 75}</pre>

## **4. 문자열**

### 연습문제 1

- 변수 `a`를 활용하여 문자열 포맷팅으로 출력하세요
- print() 를 사용합니다.

In [72]:
a = 1234

In [73]:
# 코드를 입력해 주세요
print('{0}의 값은 {1}입니다'.format('a', a))

a의 값은 1234입니다


<p><strong>[출력 결과]</strong></p><pre>a의 값은 1234입니다
</pre>

- 변수 `a`, `b`를 활용하여 문자열 포맷팅으로 출력하세요
- 단, `a`, `b`의 값은 소수 3째짜리까지 출력하고, 그 이하는 반올림합니다.

In [74]:
a = 3.141592
b = 6.181112

In [75]:
# 코드를 입력해 주세요
formatted_a = round(a, 3)
formatted_b = round(b, 3)
print(f"a의 값은 {formatted_a:.3f}, b의 값은 {formatted_b:.3f}입니다")

a의 값은 3.142, b의 값은 6.181입니다


<p><strong>[출력 결과]</strong></p><pre>a의 값은 3.142, b의 값은 6.181입니다
</pre>

### 연습문제 2

문자열 "210101-1234567"을 인덱싱 하여 성별을 나타내는 숫자를 출력 하세요

In [76]:
registration_number =  '210101-1234567'

In [79]:
# 코드를 입력해 주세요
print(registration_number[7])

1


<p><strong>[출력 결과]</strong></p><pre>'1'</pre>

### 연습문제 3

문자열 '20210101Sunny'를 슬라이싱하여 날짜는 `date`변수에 날씨는 `weather` 변수에 저장 후 출력 하세요

In [81]:
sample = '20210101Sunny'

In [82]:
# 코드를 입력해 주세요
date = sample[:8]  # 처음부터 8번 인덱스 직전까지 (20210101)
weather = sample[8:]  # 8번 인덱스부터 끝까지 (Sunny)

# 변수에 저장된 값을 출력
print(f"date: {date}, weather: {weather}")

date: 20210101, weather: Sunny


<p><strong>[출력 결과]</strong></p><pre>date: 20210101, weather: Sunny
</pre>

(문자열 슬라이싱을 활용)자동차 번호가 다음과 같을 때 뒤에 4자리만 출력하세요.

In [83]:
license_plate = "13나 5645"

In [84]:
# 코드를 입력해 주세요
backNum = license_plate[4:]
print(backNum)

5645


<p><strong>[출력 결과]</strong></p><pre>5645
</pre>

### 연습문제 4

- 다음 중 a, b, c에서 공통으로 사용된 알파벳(공백 포함)을 출력하세요.
- list 형식으로 출력하세요.

In [85]:
a = 'Hello, World'
b = 'World To World'
c = 'world to world'

In [90]:
# 대소문자를 구분하여 문자열에서 모든 문자(알파벳 및 공백)을 추출하고 집합으로 변환
set_a = set(a)
set_b = set(b)
set_c = set(c)

# 공통으로 사용된 모든 문자를 찾기 위해 교집합을 구함
common_characters = list(set_a & set_b & set_c)

# 결과를 정렬하여 리스트로 출력
common_characters.sort()
print(common_characters)

[' ', 'd', 'l', 'o', 'r']


<p><strong>[출력 결과]</strong></p><pre>[' ', 'd', 'l', 'o', 'r']</pre>

### 연습문제 5

sample = ['Python', 'is', 'too', 'interesting']

리스트를 Python is too interesting 이라는 문자열로 출력하세요

In [91]:
sample = ['Python', 'is', 'too', 'interesting']

In [93]:
# 코드를 입력해 주세요

result = ' * '.join(sample)
result = "'" + result + "'"
print(result)

'Python * is * too * interesting'


<p><strong>[출력 결과]</strong></p><pre>'Python * is * too * interesting'</pre>

In [94]:
phone_number = ['010', '1234', '5678']

In [95]:
# 코드를 입력해 주세요
result = '-'.join(phone_number)
result = "'" + result + "'"
print(result)

'010-1234-5678'


<p><strong>[출력 결과]</strong></p><pre>'010-1234-5678'</pre>

## **5. 연산**

### 연습문제

`a`와 `b`를 덧셈하여 다음의 결과가 나오도록 코드를 입력해 주세요

- 단, `a`와 `b`의 값을 수정하지 않습니다.

In [103]:
a = '10'
b = 20

In [98]:
# 코드를 입력해 주세요
print(int(a) + b)

30


<p><strong>[출력 결과]</strong></p><pre>30</pre>

In [107]:
# 코드를 입력해 주세요
print(a + str(b))

1020


<p><strong>[출력 결과]</strong></p><pre>'1020'</pre>

## **6. 비교/논리/삼항연산자, ⭐⭐조건문**

### 연습문제 1

사용자에게 몸무게를 입력 받아 과체중인지 정상 체중인지 출력하는 프로그램을 만드세요.

- 몸무게가 100kg 초과이면 '귀하는 과체중 입니다.' 메시지를 출력하세요
- 몸무게가 100kg 이하이면 '귀하는 정상체중 입니다.' 메시지를 출력하세요
- 몸무게 입력은 input() 함수를 사용하여 입력 받습니다.
- 반드시 **삼항연산자를 활용**하세요

In [111]:
# 입력 예시 (105로 입력한 경우)
weight = input('몸무게를 입력해 주세요: ')

몸무게를 입력해 주세요: 43


In [110]:
# 코드를 입력해 주세요
if int(weight) > 100:
  print('귀하는 과체중 입니다.')
else:
  print('귀하는 정상체중 입니다.')

귀하는 정상체중 입니다.


<p><strong>[출력 결과]</strong></p><pre>'귀하는 과체중 입니다.'</pre>

### 연습문제 2

#### Q1

- (몸무게 + 100 - 키)를 비만 수치 공식이라고 가정합니다.
- 키와 몸무게를 입력받아 첫 번째 줄에 비만 수치를 출력합니다.
- 비만 수치가 0보다 크면 비만이라는 메시지를 출력하세요.

```
입력 예) info =[155, 60] (키, 몸무게 순)
출력[1]: 비만수치는 5 입니다.
출력[2]: 당신은 비만입니다.
```

In [112]:
info = [155, 60]

In [113]:
# 코드를 입력해 주세요
one = info[1] + 100 - info[0]
print(f'비만수치는 {one}입니다')

if one > 0:
  print('당신은 비만입니다.')

비만수치는 5입니다
당신은 비만입니다.


<p><strong>[출력 결과]</strong></p><pre>비만수치는 5입니다
당신은 비만입니다.
</pre>

#### Q2

성적을 입력받아 90점 이상이면 A, 80점이상이면 B, 70점 이상이면 C, 70점 미만이면 D를 출력하세요
- 단, 점수가 음수이거나 100점을 초과하는 경우에는 '에러'를 출력하세요

힌트) input() 함수를 사용합니다.
- input() 함수는 사용자로부터 입력 값을 받을 수 있습니다.

In [115]:
# 코드를 입력해 주세요
score = int(input('점수를 입력해 주세요: '))
if (score < 0) | (score > 100):
  print('에러')
elif score >= 90:
  print('A')
elif score >= 80:
  print('B')
elif score >= 70:
  print('C')
else:
  print('D')

점수를 입력해 주세요: 45
D


<p><strong>[출력 결과]</strong></p><pre>점수를 입력해 주세요: 95
A
</pre>

- 95점 입력시: A
- 75점 입력시: C
- -10점 입력시: 에러
- 105점 입력시: 에러

#### Q3

사용자로부터 `달러`, `엔`, `유로`, 또는 `위안` 금액을 입력받은 후 이를 `원`으로 변환하는 프로그램을 작성해 주세요.

각 통화별 환율은 다음과 같다.

- 달러: 1237.0
- 엔: 105.4
- 유로: 1349.69
- 위안: 194.57

사용자는 10 달러, 200 엔, 30 유로, 100 위안과 같이 금액과 통화명 사이에 **공백**을 넣어 입력한다고 가정합니다.

입력 예시) `10 달러`

결과 예시) `11670.0원`

In [117]:
exchange_rates = {
    '달러': 1237.0,
    '엔': 105.4,
    '유로': 1349.69,
    '위안': 194.57
}

user_input = input("입력: ")
# 입력 문자열을 공백으로 분리하여 금액과 통화명 추출
amount, currency = user_input.split()

amount = float(amount)

# 입력된 통화명이 환율 정보에 있는지 확인하고 변환 계산
if currency in exchange_rates:
    exchange_rate = exchange_rates[currency]
    result = amount * exchange_rate
    print(f"{result:.1f} 원")

입력: 10 달러
12370.0 원


<p><strong>[출력 결과]</strong></p><pre>입력: 10 달러
12370.0 원
</pre>

## **⭐⭐ 7. 반복문**

### 연습문제 1

반복문과 조건문을 활용하여 `sample` 에서 음수 값만 출력해 주세요

In [124]:
sample = [1, 11, 22, -20, 55, -33, 0, -5, 12, 35, 19, -9, -1, -100, 90]

In [125]:
# 코드를 입력해 주세요
for i in sample:
  if i < 0:
    print(i)

-20
-33
-5
-9
-1
-100


<p><strong>[출력 결과]</strong></p><pre>-20
-33
-5
-9
-1
-100
</pre>

반복문과 조건문을 활용하여 `sample`의 요소 중 10보다 크고 50보다 작은 요소만 출력해 주세요

In [127]:
# 코드를 입력해 주세요
for i in sample:
  if (i > 10) & (i < 50):
    print(i)

11
22
12
35
19


<p><strong>[출력 결과]</strong></p><pre>11
22
12
35
19
</pre>

### 연습문제 2

다음의 문자열을 **반복문**을 활용하여 출력하되, **모음('a', 'e', 'i', 'o', 'u')만 대문자**로 자음은 소문자로 출력해 주세요.

In [128]:
sample = "PyThoN Is fUn"

In [132]:
# 코드를 입력해 주세요
#     if i.lower() in ['a', 'e', 'i', 'o', 'u']:
for i in sample:
  if i.lower() in ['a', 'e', 'i', 'o', 'u']:
    print('(모음) ' + i.upper())
  elif i == ' ':
    print('(공백)')
  else:
    print(i.lower())

p
y
t
h
(모음) O
n
(공백)
(모음) I
s
(공백)
f
(모음) U
n


<p><strong>[출력 결과]</strong></p><pre> p
 y
 t
 h
(모음) O
 n
(공백)
(모음) I
 s
(공백)
 f
(모음) U
 n
</pre>

### 연습문제 3

- 구구단을 반복문을 사용하여 만들어 보세요!
- 구구단은 2단 부터 9단까지 있습니다.

In [135]:
# 코드를 입력해 주세요
for i in range(2, 10):
    for j in range(1, 10):
        result = i * j
        print(f"{i} x {j} = {result}")
    print()


2 x 1 = 2
2 x 2 = 4
2 x 3 = 6
2 x 4 = 8
2 x 5 = 10
2 x 6 = 12
2 x 7 = 14
2 x 8 = 16
2 x 9 = 18

3 x 1 = 3
3 x 2 = 6
3 x 3 = 9
3 x 4 = 12
3 x 5 = 15
3 x 6 = 18
3 x 7 = 21
3 x 8 = 24
3 x 9 = 27

4 x 1 = 4
4 x 2 = 8
4 x 3 = 12
4 x 4 = 16
4 x 5 = 20
4 x 6 = 24
4 x 7 = 28
4 x 8 = 32
4 x 9 = 36

5 x 1 = 5
5 x 2 = 10
5 x 3 = 15
5 x 4 = 20
5 x 5 = 25
5 x 6 = 30
5 x 7 = 35
5 x 8 = 40
5 x 9 = 45

6 x 1 = 6
6 x 2 = 12
6 x 3 = 18
6 x 4 = 24
6 x 5 = 30
6 x 6 = 36
6 x 7 = 42
6 x 8 = 48
6 x 9 = 54

7 x 1 = 7
7 x 2 = 14
7 x 3 = 21
7 x 4 = 28
7 x 5 = 35
7 x 6 = 42
7 x 7 = 49
7 x 8 = 56
7 x 9 = 63

8 x 1 = 8
8 x 2 = 16
8 x 3 = 24
8 x 4 = 32
8 x 5 = 40
8 x 6 = 48
8 x 7 = 56
8 x 8 = 64
8 x 9 = 72

9 x 1 = 9
9 x 2 = 18
9 x 3 = 27
9 x 4 = 36
9 x 5 = 45
9 x 6 = 54
9 x 7 = 63
9 x 8 = 72
9 x 9 = 81



<p><strong>[출력 결과]</strong></p><pre>2 X 1 = 2
2 X 2 = 4
2 X 3 = 6
2 X 4 = 8
2 X 5 = 10
2 X 6 = 12
2 X 7 = 14
2 X 8 = 16
2 X 9 = 18

3 X 1 = 3
3 X 2 = 6
3 X 3 = 9
3 X 4 = 12
3 X 5 = 15
3 X 6 = 18
3 X 7 = 21
3 X 8 = 24
3 X 9 = 27

4 X 1 = 4
4 X 2 = 8
4 X 3 = 12
4 X 4 = 16
4 X 5 = 20
4 X 6 = 24
4 X 7 = 28
4 X 8 = 32
4 X 9 = 36

5 X 1 = 5
5 X 2 = 10
5 X 3 = 15
5 X 4 = 20
5 X 5 = 25
5 X 6 = 30
5 X 7 = 35
5 X 8 = 40
5 X 9 = 45

6 X 1 = 6
6 X 2 = 12
6 X 3 = 18
6 X 4 = 24
6 X 5 = 30
6 X 6 = 36
6 X 7 = 42
6 X 8 = 48
6 X 9 = 54

7 X 1 = 7
7 X 2 = 14
7 X 3 = 21
7 X 4 = 28
7 X 5 = 35
7 X 6 = 42
7 X 7 = 49
7 X 8 = 56
7 X 9 = 63

8 X 1 = 8
8 X 2 = 16
8 X 3 = 24
8 X 4 = 32
8 X 5 = 40
8 X 6 = 48
8 X 7 = 56
8 X 8 = 64
8 X 9 = 72

9 X 1 = 9
9 X 2 = 18
9 X 3 = 27
9 X 4 = 36
9 X 5 = 45
9 X 6 = 54
9 X 7 = 63
9 X 8 = 72
9 X 9 = 81

</pre>

### 연습문제 4

사용자의 입력이 'MALE', 'FEMALE'인 경우는 while 문을 종료하고, 그렇지 않으면 "잘못된 입력 값입니다."를 출력하세요

- input() 함수로 사용자로부터 입력을 받습니다.

In [139]:
# 코드를 입력해 주세요
while True:
  gender = input()
  if (gender == 'MALE') | (gender == 'FEMALE'):
    print('====\n프로그램 종료')
    break
  else:
    print('잘못된 입력 값입니다.')

male
잘못된 입력 값입니다.
animal
잘못된 입력 값입니다.
FEMALE
====
프로그램 종료


<p><strong>[출력 결과]</strong></p><pre>male
잘못된 입력 값입니다.
animal
잘못된 입력 값입니다.
FEMALE
====
프로그램 종료
</pre>

### 연습 문제 4

1부터 50까지의 숫자 중에서 짝수의 합을 구하세요 (while 사용)

In [142]:
# 코드를 입력해 주세요
count = 0
total = 0

while count < 50:
    count += 2
    total += count
    print(f"count: {count}, total: {total}")

count: 2, total: 2
count: 4, total: 6
count: 6, total: 12
count: 8, total: 20
count: 10, total: 30
count: 12, total: 42
count: 14, total: 56
count: 16, total: 72
count: 18, total: 90
count: 20, total: 110
count: 22, total: 132
count: 24, total: 156
count: 26, total: 182
count: 28, total: 210
count: 30, total: 240
count: 32, total: 272
count: 34, total: 306
count: 36, total: 342
count: 38, total: 380
count: 40, total: 420
count: 42, total: 462
count: 44, total: 506
count: 46, total: 552
count: 48, total: 600
count: 50, total: 650


<p><strong>[출력 결과]</strong></p><pre>count: 2, total: 2
count: 4, total: 6
count: 6, total: 12
count: 8, total: 20
count: 10, total: 30
count: 12, total: 42
count: 14, total: 56
count: 16, total: 72
count: 18, total: 90
count: 20, total: 110
count: 22, total: 132
count: 24, total: 156
count: 26, total: 182
count: 28, total: 210
count: 30, total: 240
count: 32, total: 272
count: 34, total: 306
count: 36, total: 342
count: 38, total: 380
count: 40, total: 420
count: 42, total: 462
count: 44, total: 506
count: 46, total: 552
count: 48, total: 600
count: 50, total: 650
</pre>

### 연습 문제 5

**while**을 이용하여 다음과 같이 @ 찍기에 도전해 보세요!

In [152]:
# 코드를 입력해 주세요
count = 1

while count <= 5:
    print('@' * count)
    count += 1


@
@@
@@@
@@@@
@@@@@


<p><strong>[출력 결과]</strong></p><pre>@
@@
@@@
@@@@
@@@@@
</pre>

In [1]:
# 코드를 입력해 주세요
count = 1

while count <= 3:
    print('@' * count)
    count += 1

num = 2
while num >= 1:
    print('@' * num)
    num -= 1

@
@@
@@@
@@
@


<p><strong>[출력 결과]</strong></p><pre>@
@@
@@@
@@
@
</pre>

## **⭐ 8. Comprehension**
- 별도의 연습문제는 없지만 중요한 파트이니 실습파일 공부하시는 걸 !적극! 추천드립니다😊

## **⭐⭐⭐ 9. 함수**

### 연습문제 1

알파벳 문자열을 인수로 넣으면  함수를 구현하세요

- 함수의 이름은 `capital_letter` 로 선언합니다.
- 함수는 **letter**라는 위치인수를 가집니다.
- 함수는 **letter**인수에 받은 문자열을 모두 대문자로 바꾸고, 바꾼 값을 `return` 합니다.

In [121]:
# 코드를 입력해 주세요
def capital_letter(letter):
  return letter.upper()

print(capital_letter('my name is elsa'))

MY NAME IS ELSA


<p><strong>[출력 결과]</strong></p><pre>'MY NAME IS ELSA'</pre>

### 연습문제 2

def calc_many(choice, *args) 함수에서
- choice가 `sum` 이면 **합**을 구해서 return 합니다.
- choice 가 `mul` 이면 **곱**한 결과를 return 합니다.

함수를 구현하세요

In [122]:
# 코드를 입력해 주세요
def calc_many(choice, *args):
    if choice == 'sum':
        result = sum(args)
    elif choice == 'mul':
        result = 1
        for num in args:
            result *= num
    else:
        return "지원하지 않는 연산입니다."

    return result

In [123]:
# 결과: 10
print(calc_many('sum', 0, 1, 2, 3, 4))
# 결과: 120
print(calc_many('mul', 1, 2, 3, 4, 5))

10
120


## **10. 파이썬 내장함수**

### 연습문제

`zip()`과 반복문을 활용하여 다음과 같이 출력해 주세요

In [119]:
fruit = ['사과', '자두', '딸기', '망고']
price = [1000, 3000, 2500, 6000]

In [120]:
# 코드를 입력해 주세요
for f, p in zip(fruit, price):
    print(f"({f} / {p} 원)")

(사과 / 1000 원)
(자두 / 3000 원)
(딸기 / 2500 원)
(망고 / 6000 원)


<p><strong>[출력 결과]</strong></p><pre>(사과 / 1000 원)
(자두 / 3000 원)
(딸기 / 2500 원)
(망고 / 6000 원)
</pre>